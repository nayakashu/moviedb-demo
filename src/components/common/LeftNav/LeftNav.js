import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { closeLeftNav } from '../../../actions/navActions';
import './LeftNav.scss';

export class LeftNav extends Component {
  render() {
    return (
      <div className="left-nav-container">
        <i
          className="material-icons left-nav-close"
          onClick={this.props.closeLeftNav}
        >
          close
        </i>
        <ul className="left-nav-menu">
          <li>
            <Link to="/">
              <i className="material-icons">movie</i>Top Movies
            </Link>
          </li>
          <li>
            <Link to="/mymovies">
              <i className="material-icons">person</i> My Movies
            </Link>
          </li>
          <li>
            <i className="material-icons">settings</i>Settings
          </li>
        </ul>
      </div>
    );
  }
}

export default connect(
  null,
  { closeLeftNav }
)(LeftNav);
