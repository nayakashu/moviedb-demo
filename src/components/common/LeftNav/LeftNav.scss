@import '../../../Common.scss';

.left-nav-container {
  position: absolute;
  top: 0px;
  left: 0px;
  width: 250px;
  height: 100%;
  z-index: 300;
  background: $header-bg-color;
  color: $content-text-color;
  box-shadow: 1px 2px 5px $shadow-color-dark;

  i.left-nav-close {
    float: right;
    margin: 20px;
    cursor: pointer;
  }

  ul.left-nav-menu {
    margin: 10px;
    margin-top: 60px;
    list-style: none;
    li {
      padding: 10px 20px;
      font-size: 16px;
      cursor: pointer;
      a {
        color: $content-text-color;
        &:hover {
          text-decoration: none;
        }
      }

      i {
        font-size: 19px;
        margin-right: 13px;
        position: relative;
        top: 3px;
      }
    }
  }
}
