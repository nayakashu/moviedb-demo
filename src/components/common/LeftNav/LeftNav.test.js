import React from 'react';
import { shallow } from 'enzyme';
import { LeftNav } from './LeftNav';

describe('LeftNav', () => {
  const leftNav = shallow(<LeftNav />);

  it('renders correctly', () => {
    expect(leftNav).toMatchSnapshot();
  });
});
