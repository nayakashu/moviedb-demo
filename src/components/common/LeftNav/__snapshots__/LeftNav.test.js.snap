// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftNav renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LeftNav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <i
          className="material-icons left-nav-close"
        >
          close
        </i>,
        <ul
          className="left-nav-menu"
        >
          <li>
            <i
              className="material-icons"
            >
              movie
            </i>
            Top Movies
          </li>
          <li>
            <i
              className="material-icons"
            >
              person
            </i>
             My Movies
          </li>
          <li>
            <i
              className="material-icons"
            >
              settings
            </i>
            Settings
          </li>
        </ul>,
      ],
      "className": "left-nav-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "close",
          "className": "material-icons left-nav-close",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "close",
        "type": "i",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <i
                className="material-icons"
              >
                movie
              </i>
              Top Movies
            </li>,
            <li>
              <i
                className="material-icons"
              >
                person
              </i>
               My Movies
            </li>,
            <li>
              <i
                className="material-icons"
              >
                settings
              </i>
              Settings
            </li>,
          ],
          "className": "left-nav-menu",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="material-icons"
                >
                  movie
                </i>,
                "Top Movies",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "movie",
                  "className": "material-icons",
                },
                "ref": null,
                "rendered": "movie",
                "type": "i",
              },
              "Top Movies",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="material-icons"
                >
                  person
                </i>,
                " My Movies",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "person",
                  "className": "material-icons",
                },
                "ref": null,
                "rendered": "person",
                "type": "i",
              },
              " My Movies",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="material-icons"
                >
                  settings
                </i>,
                "Settings",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "settings",
                  "className": "material-icons",
                },
                "ref": null,
                "rendered": "settings",
                "type": "i",
              },
              "Settings",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <i
            className="material-icons left-nav-close"
          >
            close
          </i>,
          <ul
            className="left-nav-menu"
          >
            <li>
              <i
                className="material-icons"
              >
                movie
              </i>
              Top Movies
            </li>
            <li>
              <i
                className="material-icons"
              >
                person
              </i>
               My Movies
            </li>
            <li>
              <i
                className="material-icons"
              >
                settings
              </i>
              Settings
            </li>
          </ul>,
        ],
        "className": "left-nav-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "close",
            "className": "material-icons left-nav-close",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "close",
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <i
                  className="material-icons"
                >
                  movie
                </i>
                Top Movies
              </li>,
              <li>
                <i
                  className="material-icons"
                >
                  person
                </i>
                 My Movies
              </li>,
              <li>
                <i
                  className="material-icons"
                >
                  settings
                </i>
                Settings
              </li>,
            ],
            "className": "left-nav-menu",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="material-icons"
                  >
                    movie
                  </i>,
                  "Top Movies",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "movie",
                    "className": "material-icons",
                  },
                  "ref": null,
                  "rendered": "movie",
                  "type": "i",
                },
                "Top Movies",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="material-icons"
                  >
                    person
                  </i>,
                  " My Movies",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "person",
                    "className": "material-icons",
                  },
                  "ref": null,
                  "rendered": "person",
                  "type": "i",
                },
                " My Movies",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="material-icons"
                  >
                    settings
                  </i>,
                  "Settings",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "settings",
                    "className": "material-icons",
                  },
                  "ref": null,
                  "rendered": "settings",
                  "type": "i",
                },
                "Settings",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
